pkg/skaffold/runner/runner.go

```diff
@@ -92,23 +72,18 @@ func NewForConfig(opts *config.SkaffoldOptions, cfg *config.SkaffoldConfig) (*Sk
 		return nil, errors.Wrap(err, "parsing skaffold deploy config")
 	}
 
+	deployer = label.WithLabels(deployer, labels.Merge(opts, builder, deployer, tagger))
```

Comment:
This will greedily evaluate the labels for each component, which means any labels added during the run won't get added (e.g. profiles when we add labels for those). Can we call `labels.Merge()` right before we actually apply the labels?

Reply:
Makes sense. Let me fix that